src/Lean/Elab/Command.lean

```diff
@@ -12,17 +12,35 @@ import Lean.Language.Basic
 
 namespace Lean.Elab.Command
 
+/--
+A `Scope` collects context information which is inherently scoped: this contains
+the current namespace and information about `open`, `universe` or `variable` declarations.
+There is always a base scope; scopes can be nested.
+
+`universe`, `open` and `variable` declarations only modify the current scope
+(and all scopes nested below it): these declarations usually *append* to the
+context in the current scope; `variable` statements can also change existing items.
+(XXX Is shadowing possible? If so, this sentence should be reworded.)
+Unclosed scopes are automatically closed at the end of the current module.
+-/
 structure Scope where
   header        : String
   opts          : Options := {}
+  /-- The current namespace: by default, this is an anonymous namespace -/
   currNamespace : Name := Name.anonymous
+  /-- All currently `open`ed namespaces -/
```

Comment:
```suggestion
  /-- All currently `open`ed namespaces and names. -/
```
The `OpenDecl` type gives information about individual names too, needed to support `open MyNamespace (decl1, decl2)`.